cli/flox/src/commands/services/start.rs

```diff
@@ -0,0 +1,270 @@
+use crate::subcommand_metric;
+use crate::utils::message;
+
+#[derive(Bpaf, Debug, Clone)]
+pub struct Start {
+    #[bpaf(external(environment_select), fallback(Default::default()))]
+    environment: EnvironmentSelect,
+
+    /// Names of the services to start
+    #[bpaf(positional("name"))]
+    names: Vec<String>,
+}
+
+impl Start {
+    #[instrument(name = "start", skip_all)]
+    pub async fn handle(self, config: Config, flox: Flox) -> Result<()> {
+        subcommand_metric!("services::start");
+
+        let mut concrete_environment = supported_concrete_environment(&flox, &self.environment)?;
+        let activated_environments = activated_environments();
+
+        if !activated_environments.is_active(&UninitializedEnvironment::from_concrete_environment(
+            &concrete_environment,
+        )?) {
+            return Err(anyhow!(indoc! {"
+                Cannot start services for an environment that is not activated.
+
+                To activate and start services, run 'flox activate -s'
+            "}));
+        }
+
+        // TODO: this doesn't need to be mut
+        let env = concrete_environment.dyn_environment_ref_mut();
+        let socket = env.services_socket_path(&flox)?;
+
+        let start_new_process_compose = if !socket.exists() {
+            true
+        } else {
+            let processes = ProcessStates::read(&socket)?;
+            let all_processes_stopped = processes.iter().all(|p| p.is_stopped());
+            if all_processes_stopped {
+                process_compose_down(&socket)?;
+            }
+            all_processes_stopped
+        };
+
+        if start_new_process_compose {
+            debug!("starting services in new process-compose instance");
+            self.start_with_new_process_compose(config, flox, concrete_environment)
+                .await
+        } else {
+            debug!("starting services with existing process-compose instance");
+            Self::start_with_existing_process_compose(socket, &self.names, &mut stderr())
+        }
+    }
+
+    /// Note that this must be called within an existing activation, otherwise it
+    /// will leave behind a process-compose since it doesn't start a watchdog.
+    async fn start_with_new_process_compose(
+        &self,
+        config: Config,
+        flox: Flox,
+        mut concrete_environment: ConcreteEnvironment,
+    ) -> Result<()> {
+        let environment = concrete_environment.dyn_environment_ref_mut();
+        let lockfile = environment.lockfile(&flox)?;
+        let LockedManifest::Catalog(lockfile) = lockfile else {
+            unreachable!("at least it should be after https://github.com/flox/flox/issues/1858")
+        };
+        for name in &self.names {
+            if !lockfile.manifest.services.contains_key(name) {
+                return Err(service_does_not_exist_error(name));
+            }
+        }
+        Activate {
+            environment: self.environment.clone(),
+            trust: false,
```

Comment:
Adding a comment:
```
            // We currently only check for trust for remote environments,
            // but set this to false in case that changes.
```